### PR TITLE
Save processed Dataset

### LINKED ISSUE #96: Save processed datasets
Datasets are currently re-processed every time as mlx arrays cannot be pickled currently.

### PR DESCRIPTION
**Pull request by francescofarina**
## Proposed changes

Implementing `save` and `load` for datasets with `pickle`.

Closes #96.

## Checklist

Put an `x` in the boxes that apply.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the necessary documentation

### CODE REVIEW

File: mlx_graphs/datasets/superpixel.py [hunk 1]
```diff
@@ -168,6 +168,15 @@ def _img_size(self):
             return 28
         return 32
 
+    @property
+    def processed_path(self) -> str:
+        # processed path includes split and ude_features
```
Comment (TristanBilot) [inline mlx_graphs/datasets/superpixel.py:173]:
```suggestion
        # processed path includes split and use_features
```

**Review by TristanBilot — COMMENTED** [review]
Nice work! Maybe shall we wait that mlx 0.6.0 is released before merging?

**Comment by francescofarina** [comment]
I think it's fair to assume that our main branch is in sync with the `mlx` main branch - we'll wait for `mlx` 0.6 before making the new release though

**Comment by francescofarina** [comment]
0.6 is out anyway 😉

**Comment by TristanBilot** [comment]
Amazing!

**Pull request merged**